### PR TITLE
DOC: Improved docs for `jnp.fft.fftn`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/fft.py [hunk 1]
```diff
@@ -109,10 +109,76 @@ def _fft_core(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
 def fftn(a: ArrayLike, s: Shape | None = None,
          axes: Sequence[int] | None = None,
          norm: str | None = None) -> Array:
+  r"""Compute the multidimensional discrete fourier transform.
+
+  JAX implementation of :func:`numpy.fft.fftn`.
+
+  Args:
+    a: input array
+    s: integer or sequence of integers. Specifies the shape of the result. If not
```
Comment (dfm) [inline jax/_src/numpy/fft.py:121]: I think a sequence is required here?

```suggestion
    s: sequence of integers. Specifies the shape of the result. If not
```

File: jax/_src/numpy/fft.py [hunk 2]
```diff
@@ -109,10 +109,76 @@ def _fft_core(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
   return transformed
 
 
-@implements(np.fft.fftn)
 def fftn(a: ArrayLike, s: Shape | None = None,
          axes: Sequence[int] | None = None,
          norm: str | None = None) -> Array:
+  r"""Compute the multidimensional discrete fourier transform.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:115]:
```suggestion
  r"""Compute the multidimensional discrete Fourier transform.
```

"Fourier" should be capitalized. But also, I wasn't immediately clear on what "multidimensional" meant here. Maybe we should clarify that it is the FFT of a multidimensional array along specified axes or something?

File: jax/_src/numpy/fft.py [hunk 3]
```diff
@@ -109,10 +109,76 @@ def _fft_core(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+  JAX implementation of :func:`numpy.fft.fftn`.
+
+  Args:
+    a: input array
+    s: integer or sequence of integers. Specifies the shape of the result. If not
+      specified, it will default to the shape of ``a`` along the specified ``axes``.
+    axes: integer or sequence of integers, default=None. Specifies the axes along
+      which the transform is computed.
+    norm: string. The normalization mode. "backward", "ortho" and "forward" are
+      supported.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:126]: It would be nice to describe what the different normalizations mean somewhere. We don't currently and I don't think it needs to be part of this PR, but I wanted to note that point while I remembered it!

**Review by dfm — CHANGES_REQUESTED** [review]
Looks great - thanks! I've requested a couple of small changes.

File: jax/_src/numpy/fft.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py:115]: "Compute the FFT of a multidimensional array along specified axes" sounds good to me.

**Comment by rajasekharporeddy** [comment]
Hi @dfm, implemented the changes suggested and squashed into single commit.

File: jax/_src/numpy/fft.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

File: jax/_src/numpy/fft.py [hunk 4]
```diff
@@ -109,10 +109,77 @@ def _fft_core(func_name: str, fft_type: xla_client.FftType, a: ArrayLike,
...
+  r"""Compute discrete Fourier transform of multidimensional array along given axes.
+
+  JAX implementation of :func:`numpy.fft.fftn`.
+
+  Args:
+    a: input array
+    s: sequence of integers. Specifies the shape of the result. If not specified,
+      it will default to the shape of ``a`` along the specified ``axes``.
+    axes: integer or sequence of integers, default=None. Specifies the axes along
+      which the transform is computed.
```
Comment (dfm) [inline jax/_src/numpy/fft.py:124]: Sorry, I should have caught this before, but `axes` also needs to always be a sequence as far as I can tell.

**Review by dfm — CHANGES_REQUESTED** [review]

File: jax/_src/numpy/fft.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Done

**Review by dfm — APPROVED** [review]

File: jax/_src/numpy/fft.py (hunk 2 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: Hi @dfm, I think here "multidimensional discrete Fourier transform" seems to be the correct description. because even `jnp.fft.fft` computes the discrete Fourier transform of multidimensional array. 

`jnp.fft.fftn` computes FFT along multiple axes (hence called multidimensional FFT) whereas `jnp.fft.fftn` computes FFT along only one axis (hence called one-dimensional FFT). Could you please tell me whether my understanding is correct?

Comment (dfm) [reply jax/_src/numpy/fft.py]: I think you're right! I do find the terminology a bit confusing, but I think you're right that something like what you're suggesting would be more accurate. Would you be willing to open a new PR?

Comment (rajasekharporeddy) [reply jax/_src/numpy/fft.py]: I would be writing the docstrings for `jnp.fft.fft` and `jnp.fft.ifft`. I will include this change in that PR. Is that fine?

Comment (dfm) [reply jax/_src/numpy/fft.py]: Sounds good to me!
